hbase-client/src/main/java/org/apache/hadoop/hbase/HRegionInfo.java

```diff
@@ -964,8 +965,7 @@ public static HRegionInfo parseFrom(final DataInputStream in) throws IOException
     }
 
     //assumption: if Writable serialization, it should be longer than pblen.
-    int read = in.read(pbuf);
-    if (read != pblen) throw new IOException("read=" + read + ", wanted=" + pblen);
+    IOUtils.readFully(in, pbuf, 0, pblen);
```

Comment:
DataInputStream has a readFully method, so I think here we could just use that one?

Reply:
Let me do your suggestion instead. Thanks.